app/controllers/courses_controller.rb

```diff
@@ -198,6 +198,13 @@ def manage_exercises
     @exercises_id_map = @exercises.map { |e| [e.id, e] }.to_h
   end
 
+  def toggle_submission_result_visibility
+    authorize! :toggle_submission_result_visibility, @course
+    @course.hide_submission_result = !@course.hide_submission_result
+    @course.save!
```

Comment:
Nit: this could be a method in model...
```ruby
def toggle_submission_result_visiblity
  self.hide_submission_results = !self.hide_submission_results
end
```